src/sparse/impl/KokkosSparse_spmv_struct_impl.hpp

```diff
@@ -527,98 +504,156 @@ struct SPMV_Struct_Functor {
       });
   }
 
+  void compute_exterior(const int64_t worksets, const int team_size, const int vector_length) {
```

Comment:
Does the interior and exterior have to be two different calls, kernels launches ? Can this be two device callable functions ?

Reply:
One way to handle this would be to loop over all the rows of the matrix and to apply the different algorithms depending on the row index but that would lead to significant branch divergence on GPU.
Otherwise using a CUDA graph to launch the two kernels might be possible?
My goal is to eventually separate the interior and exterior further to allow for computation/communication overlap.

Let me know what you think?

Reply:
I didn't think of computation/communication overlap. Good point. I hope launch overheads become so bad where we cannot overlap.